internal/control_plane/server/api.go

```diff
@@ -6,15 +6,40 @@ import (
 	"octavius/internal/control_plane/config"
 	"octavius/internal/control_plane/db/etcd"
 	"octavius/internal/control_plane/logger"
-	"octavius/internal/control_plane/server/metadata/repository"
+	executorRepo "octavius/internal/control_plane/server/repository/executor"
+	metadataRepo "octavius/internal/control_plane/server/repository/metadata"
+	"os"
+	"os/signal"
+	"syscall"
 	"time"
 
 	"octavius/internal/control_plane/server/execution"
-	protobuf "octavius/internal/pkg/protofiles/client_CP"
+	clientCPproto "octavius/internal/pkg/protofiles/client_CP"
+	executorCPproto "octavius/internal/pkg/protofiles/executor_CP"
 
 	"google.golang.org/grpc"
 )
 
+func startClientCPServer(listener net.Listener, clientServer clientCPproto.ClientCPServicesServer) {
+	server := grpc.NewServer()
+	clientCPproto.RegisterClientCPServicesServer(server, clientServer)
+	logger.Info(fmt.Sprintln("Started client server at port: ", listener.Addr().String()))
+	err := server.Serve(listener)
+	if err != nil {
+		logger.Fatal("Failed to start Client Server")
```

Comment:
This fatal would not affect the main routine since startExecutorCPServer will run inside go routine. see line #60. better use logger.Error, or return the error and handle the error outside. and also you didn't print the error cause.

I suggest return the error.